### PR TITLE
Various refactoring

### PR DESCRIPTION
**Pull request by unixcharles**
ping @jhass

### CODE REVIEW

File: spec/certificate_request_spec.rb [hunk 1]
```diff
@@ -1,22 +1,20 @@
 require 'spec_helper'
 
 describe Acme::CertificateRequest do
-  before(:all) do
-    @test_key = generate_private_key
-  end
+  let(:test_key) { generate_private_key }
```
Comment (jhass) [inline spec/certificate_request_spec.rb]: This seem to indeed make the specs on average 3 (2-4) seconds slower on my machine.

File: lib/acme/crypto.rb [hunk 2]
```diff
@@ -13,14 +13,16 @@ def generate_signed_jws(header:, payload:)
...
 
-  def jwk
-    @jwk ||= JSON::JWK.new(public_key)
-  end
-
   def thumbprint
     jwk.thumbprint
   end
 
+  private
```
Comment (jhass) [inline lib/acme/crypto.rb]: I guess we miss a spec calling this https://github.com/unixcharles/acme-client/blob/master/lib/acme/resources/challenges/dns01.rb#L14

File: spec/certificate_request_spec.rb [hunk 3]
```diff
@@ -1,22 +1,22 @@
 require 'spec_helper'
 
 describe Acme::CertificateRequest do
-  before(:all) do
-    @test_key = generate_private_key
+  def test_key
+    @test_key ||= generate_private_key
   end
```
Comment (jhass) [inline spec/certificate_request_spec.rb]: RSpec seems to do some magic for `before(:all)`, inspecting it, each example still resets the instance variable this way.

Comment (unixcharles) [inline spec/certificate_request_spec.rb]: juste moved it back into before(:all) and trimmed few more second by only generating the CSR when calling it.

**Comment by jhass** [comment]
LGTM so far :+1:

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,63 +1,19 @@
 $LOAD_PATH.unshift File.expand_path('../../lib', __FILE__)
+$LOAD_PATH.unshift File.expand_path('../support', __FILE__)
```
Comment (jhass) [inline spec/spec_helper.rb]: Oh, just realized, since we only support 2.1 and 2.2, we could use `__dir__` here and trim one `../` from each. YMMV.

Comment (unixcharles) [inline spec/spec_helper.rb]: :+1:

**Pull request merged**